source/Agents/GP/GPTrainingLoop.hpp

```diff
@@ -0,0 +1,574 @@
+
+#pragma once
+
+#include "../../core/AgentBase.hpp"
+#include "../../core/WorldBase.hpp"
+//#include "GPAgent.hpp"
+#include "GPAgent_.hpp"
+
+
+#include "CGPAgent.hpp"
+
+#include <thread>
+#include <iostream>
+#include <vector>
+#include <ranges>
+#include <cmath>
+#include <filesystem>
+#include <string>
+#include <sstream>
+#include <chrono>
+
+
+
+
+#include "tinyxml2.h"
+//#include <algorithm>
+
+
+namespace cowboys {
+
+    template<class AgentType, class EnvironmentType>
+    class GPTrainingLoop {
+    private:
+
+        std::vector<cse491::WorldBase *> environments;
+        std::vector<std::vector<cowboys::GPAgent_ *>> agents;
+
+
+        std::vector<std::vector<cse491::GridPosition>> TEMPinitialAgentPositions;
```

Comment:
I'm assuming TEMP means this will change soon? Or did the name just get left behind?